halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/GenerateCommand.java

```diff
@@ -1,4 +1,37 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.cli.command.v1;
 
-public class GenerateCommand {
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.DaemonService;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+public class GenerateCommand extends NestableCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "generate";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Generate the full Spinnaker config for you current deployment.";
+
+  @Override
+  protected void executeThis() {
+    DaemonService service = Daemon.getService();
+    String currentDeployment = service.getCurrentDeployment();
+    service.generateDeployment(currentDeployment, "", false);
```

Comment:
The current-deployment is in Halyard to allow multiple sets of Spinnaker configurations to be managed at once.

Version information is tied to the config, because the config in spinnaker/spinnaker/config is liable to change as each subcomponent changes.

What do you mean?
> the configuration you write needs to tie together, but this is where the trying would happen

Odds are, changing a version won't require rewriting your config. Some properties in the config we are extending might change, but that happens without the user knowing about it.